src/librustdoc/passes/calculate_doc_coverage.rs

```diff
@@ -67,13 +69,73 @@ impl ops::AddAssign for ItemCount {
     }
 }
 
-#[derive(Default)]
 struct CoverageCalculator {
     items: BTreeMap<FileName, ItemCount>,
+    output_format: Option<OutputFormat>,
+}
+
+fn limit_filename_len(filename: String) -> String {
+    // if a filename is too long, shorten it so we don't blow out the table
+    // FIXME(misdreavus): this needs to count graphemes, and probably also track
+    // double-wide characters...
+    if filename.len() > 35 {
+        "...".to_string() + &filename[filename.len() - 32..]
+    } else {
+        filename
+    }
+}
+
+impl ToJson for CoverageCalculator {
+    fn to_json(&self) -> Json {
+        let mut total = ItemCount::default();
+        let mut entries = BTreeMap::default();
+
+        entries.insert("files".to_owned(), Json::Array(self.items
+            .iter()
+            .filter_map(|(file, &count)| {
+                match count.percentage() {
+                    Some(percent) => Some((limit_filename_len(file.to_string()), count, percent)),
+                    None => None,
+                }
+            })
+            .map(|(name, count, percentage)| {
+                let mut fields = BTreeMap::default();
```

Comment:
```suggestion
                let fields = btreemap! {
                    "documented" => ...,
                    ...
                };
```
(from https://docs.rs/maplit/1.0.2/maplit/macro.btreemap.html -- It's already in Cargo.lock)

Reply:
Wo that's crazy!

Reply:
Ok no, it's not in librustdoc Cargo.lock file so I won't add it. However, I'll definitely keep it in mind.